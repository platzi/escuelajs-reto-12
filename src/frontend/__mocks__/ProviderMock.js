import React from 'react';
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import reducer from '../reducers';

const initialState = {
  total_price: 0,
  cart: [],
  products: [
    {
      id: '1',
      image: 'https://arepa.s3.amazonaws.com/camiseta.png',
      title: 'Camiseta',
      price: 25,
      description: 'bla bla bla bla bla',
    },
    {
      id: '3',
      image: 'https://arepa.s3.amazonaws.com/mug.png',
      title: 'Mug',
      price: 10,
      description: 'bla bla bla bla bla',
    },
    {
      id: '4',
      image: 'https://arepa.s3.amazonaws.com/pin.png',
      title: 'Pin',
      price: 4,
      description: 'bla bla bla bla bla',
    },
    {
      id: '5',
      image: 'https://arepa.s3.amazonaws.com/stickers1.png',
      title: 'Stickers',
      price: 2,
      description: 'bla bla bla bla bla',
    },
    {
      id: '6',
      image: 'https://arepa.s3.amazonaws.com/stickers2.png',
      title: 'Stickers',
      price: 2,
      description: 'bla bla bla bla bla',
    },
    {
      id: '7',
      image: 'https://arepa.s3.amazonaws.com/hoodie.png',
      title: 'Hoodie',
      price: 35,
      description: 'bla bla bla bla bla',
    },
  ],
};

const store = createStore(
  reducer,
  initialState,
  compose(applyMiddleware(thunk)),
);
const history = createBrowserHistory();

const ProviderMock = props => (
  <Provider store={store}>
    <Router history={history}>{props.children}</Router>
  </Provider>
);

export default ProviderMock;
