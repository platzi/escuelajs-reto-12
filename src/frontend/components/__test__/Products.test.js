import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import Products from '../Products';

describe('Product component', () => {
  test('Match Snapshot', () => {
    const product = create(
      <ProviderMock>
        <Products />
      </ProviderMock>,
    );
    expect(product.toJSON()).toMatchSnapshot();
  });
});
