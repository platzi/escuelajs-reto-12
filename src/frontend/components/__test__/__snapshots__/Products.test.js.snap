// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component Match Snapshot 1`] = `
<div
  className="Products"
>
  <div
    className="Products-items"
  >
    <div
      className="Products-item"
    >
      <img
        alt="Camiseta"
        src="https://arepa.s3.amazonaws.com/camiseta.png"
      />
      <div
        className="Products-item-info"
      >
        <h2>
          Camiseta
          <span>
            $
            25
          </span>
        </h2>
        <p>
          bla bla bla bla bla
        </p>
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        Comprar
      </button>
    </div>
    <div
      className="Products-item"
    >
      <img
        alt="Mug"
        src="https://arepa.s3.amazonaws.com/mug.png"
      />
      <div
        className="Products-item-info"
      >
        <h2>
          Mug
          <span>
            $
            10
          </span>
        </h2>
        <p>
          bla bla bla bla bla
        </p>
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        Comprar
      </button>
    </div>
    <div
      className="Products-item"
    >
      <img
        alt="Pin"
        src="https://arepa.s3.amazonaws.com/pin.png"
      />
      <div
        className="Products-item-info"
      >
        <h2>
          Pin
          <span>
            $
            4
          </span>
        </h2>
        <p>
          bla bla bla bla bla
        </p>
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        Comprar
      </button>
    </div>
    <div
      className="Products-item"
    >
      <img
        alt="Stickers"
        src="https://arepa.s3.amazonaws.com/stickers1.png"
      />
      <div
        className="Products-item-info"
      >
        <h2>
          Stickers
          <span>
            $
            2
          </span>
        </h2>
        <p>
          bla bla bla bla bla
        </p>
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        Comprar
      </button>
    </div>
    <div
      className="Products-item"
    >
      <img
        alt="Stickers"
        src="https://arepa.s3.amazonaws.com/stickers2.png"
      />
      <div
        className="Products-item-info"
      >
        <h2>
          Stickers
          <span>
            $
            2
          </span>
        </h2>
        <p>
          bla bla bla bla bla
        </p>
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        Comprar
      </button>
    </div>
    <div
      className="Products-item"
    >
      <img
        alt="Hoodie"
        src="https://arepa.s3.amazonaws.com/hoodie.png"
      />
      <div
        className="Products-item-info"
      >
        <h2>
          Hoodie
          <span>
            $
            35
          </span>
        </h2>
        <p>
          bla bla bla bla bla
        </p>
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        Comprar
      </button>
    </div>
  </div>
</div>
`;
